### PR TITLE
aws sqs has tags

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: plugins/aws/sqs/sqsHasTags.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'SQS Has Tags',
+    category: 'SQS',
+    domain: 'Application Integration',
+    severity: 'Low',
+    description: 'Ensures that each Amazon SQS queue has tags associated',
+    more_info: 'Tags help you to group resources together that are related to or associated with each other. It is a best practice to tag cloud resources to better organize and gain visibility into their usage.',
+    recommended_action: 'Update Amazon SQS queue and add new tags.',
```
Comment (fatima99s) [inline plugins/aws/sqs/sqsHasTags.js:11]:
```suggestion
    recommended_action: 'Update Amazon SQS queue and add tags.',
```

File: plugins/aws/sqs/sqsHasTags.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                
+                var queueName = queueUrl.split('/');
+                queueName = queueName[queueName.length-1];
+
+                var resource = `arn:${awsOrGov}:sqs:${region}:${accountId}:${queueName}`;
+
+                arnList.push(resource);
+            }
+
+            helpers.checkTags(cache, 'sqs', arnList, region, results, settings);
```
Comment (fatima99s) [inline plugins/aws/sqs/sqsHasTags.js:53]:
```suggestion
            helpers.checkTags(cache, 'SQS queue', arnList, region, results, settings);
```

File: plugins/aws/sqs/sqsHasTags.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+module.exports = {
+    title: 'SQS Has Tags',
+    category: 'SQS',
+    domain: 'Application Integration',
+    severity: 'Low',
+    description: 'Ensures that each Amazon SQS queue has tags associated',
+    more_info: 'Tags help you to group resources together that are related to or associated with each other. It is a best practice to tag cloud resources to better organize and gain visibility into their usage.',
+    recommended_action: 'Update Amazon SQS queue and add new tags.',
+    link: 'https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-queue-tags.html',
+    apis: ['SQS:listQueues', 'STS:getCallerIdentity', 'ResourceGroupsTaggingAPI:getResources', 'SQS:listQueueTags'],
```
Comment (fatima99s) [inline plugins/aws/sqs/sqsHasTags.js:13]:
```suggestion
    apis: ['SQS:listQueues', 'STS:getCallerIdentity', 'ResourceGroupsTaggingAPI:getResources'],
```

File: plugins/aws/sqs/sqsHasTags.spec.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        //     const cache = createCache(['https://sqs.us-east-1.amazonaws.com/111122223333/test-queue-1'], [
+        //         {
+        //             "ResourceARN": "arn:aws:sqs:us-east-1:111122223333:test-queue-1",
+        //             "Tags": [{ key: 'name', value: 'test-queue-1' }],
+        //         }
+        //     ]);
+        //     sqsHasTags.run(cache, {}, (err, results) => {
+        //         expect(results.length).to.equal(1);
+        //         expect(results[0].status).to.equal(0);
+        //         expect(results[0].region).to.equal('us-east-1');
```
Comment (fatima99s) [inline plugins/aws/sqs/sqsHasTags.spec.js:37]: update the message in spec file and pass check if sqs queue has tags is commented

**Review by fatima99s — CHANGES_REQUESTED** [review]

File: plugins/aws/sqs/sqsHasTags.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'SQS Has Tags',
+    category: 'SQS',
+    domain: 'Application Integration',
+    severity: 'Low',
+    description: 'Ensures that each Amazon SQS queue has tags associated',
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.js:9]:
```suggestion
    description: 'Ensures that Amazon SQS queue has tags associated.',
```

File: plugins/aws/sqs/sqsHasTags.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'SQS Has Tags',
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.js:5]:
```suggestion
    title: 'SQS Queue Has Tags',
```

File: plugins/aws/sqs/sqsHasTags.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    category: 'SQS',
+    domain: 'Application Integration',
+    severity: 'Low',
+    description: 'Ensures that each Amazon SQS queue has tags associated',
+    more_info: 'Tags help you to group resources together that are related to or associated with each other. It is a best practice to tag cloud resources to better organize and gain visibility into their usage.',
+    recommended_action: 'Update Amazon SQS queue and add tags.',
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.js:11]:
```suggestion
    recommended_action: 'Update SQS queue and add tags.',
```

File: plugins/aws/sqs/sqsHasTags.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+        async.each(regions.sqs, function(region, rcb){
+            var listQueues = helpers.addSource(cache, source,
+                ['sqs', 'listQueues', region]);
+
+            if (!listQueues) return rcb();
+
+            if (listQueues.err || !listQueues.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query for Amazon SQS queues: ${helpers.addError(listQueues)}`, region);
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.js:33]:
```suggestion
                    `Unable to query for SQS queues: ${helpers.addError(listQueues)}`, region);
```

File: plugins/aws/sqs/sqsHasTags.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                return rcb();
+            }
+
+            if (!listQueues.data.length) {
+                helpers.addResult(results, 0, 'No Amazon SQS queues found', region);
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.js:38]:
```suggestion
                helpers.addResult(results, 0, 'No SQS queues found', region);
```

File: plugins/aws/sqs/sqsHasTags.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+                return rcb();
+            }
+
+            const arnList = [];
+            for (let queueUrl of listQueues.data){
+
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.js:44]:
```suggestion
```

File: plugins/aws/sqs/sqsHasTags.js [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+            }
+
+            const arnList = [];
+            for (let queueUrl of listQueues.data){
+                
+                var queueName = queueUrl.split('/');
+                queueName = queueName[queueName.length-1];
+
+                var resource = `arn:${awsOrGov}:sqs:${region}:${accountId}:${queueName}`;
+
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.js:49]:
```suggestion
```

File: plugins/aws/sqs/sqsHasTags.spec.js [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+            });
+        });
+
+        it('should PASS if no SQS queues found', function (done) {
+            const cache = createCache([], null);
+            sqsHasTags.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].region).to.equal('us-east-1');
+                expect(results[0].message).to.include('No Amazon SQS queues found');
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.spec.js:72]:
```suggestion
                expect(results[0].message).to.include('No SQS queues found');
```

File: plugins/aws/sqs/sqsHasTags.spec.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+            });
+        });
+
+        it('should UNKNOWN if unable to query SQS queues', function (done) {
+            const cache = createCache(null, null);
+            sqsHasTags.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].region).to.equal('us-east-1');
+                expect(results[0].message).to.include('Unable to query for Amazon SQS queues');
```
Comment (alphadev4) [inline plugins/aws/sqs/sqsHasTags.spec.js:83]:
```suggestion
                expect(results[0].message).to.include('Unable to query for SQS queues');
```

**Review by alphadev4 — APPROVED** [review]

**Review by alphadev4 — APPROVED** [review]

**Pull request merged**